Assets/_Scripts/PlayerController.cs

```diff
@@ -62,33 +63,21 @@ private void Shoot()
 
     private void ActivateBullet()
     {
-        GameObject nextInactiveBullet = GetNextInactiveBullet();
+        GameObject nextInactiveBullet = bulletPool.GetPooledObject("Bullet");

```

Comment:
Strings in c# are allocated in dynamic memory - or the "heap". Every time you call this function and pass in a string like "Bullet", it'll create a new string in memory and allocate it onto the heap. Over time, this will call the garbage collector to run, which is an expensive action. Since this function will be called numerous times, it will be best to pass in a cached variable of the string.